README.md

```diff
@@ -0,0 +1,129 @@
+your allocation scripts spawn and return to resalloc, those are your
+_resources_. It can be virtual machines, docker containers, IP addresses, TODO,
+or in an extreme case even people (imagine a ticket system at the post office).
+
+There already are allocation scripts for various kinds of resources:
+
+- [resalloc-aws](https://github.com/praiskup/resalloc-aws) -
+  Allocates VMs in the Amazon EC2 cloud
+- [resalloc-ibm-cloud](https://github.com/fedora-copr/resalloc-ibm-cloud) -
+  Allocates VMs in the IBM Cloud
+- [resalloc-openstack](https://github.com/praiskup/resalloc-openstack) -
+  Allocates VMs in your OpenStack instance
+- [resalloc-kubernetes](https://github.com/TommyLike/resalloc-kubernetes) -
+  Allocates pods in your Kubernetes cluster
+
+If you can, please share your own.
+
+
+Motivation
+----------
+
+At the first sight, it may look like using resalloc is an overkill but please
+consider what resalloc can do for you and what you would eventually need to
+implement on your own when starting from scratch.
+
+- Multiple resource pools - Resalloc allows to define multiple pools of
+  resources. They can either provide identical resources or different
+  kinds of resources (e.g. x86_64 VMs from Amazon AWS and x86_64 VMs from
+  OpenStack that can be used interchangeably, and ppc64le VMs from IBM Cloud
+  that are used for a different purpose).
+- Client-server architecture - Allocating many new resources in parallel can be
+  surprisingly heavy on CPU and memory, so it can be useful to dedicate a
+  separate machine to resource allocation.
+- Gradual allocation - Allocating resources on self-hosted hardware is for free,
+  so it makes sense to use it to the full capacity at all times. However,
+  somebody will have to pay for every allocated resource in the cloud, so we
+  want to allocate only as much resources as we need at the moment.
+- Preallocation - Allocating new resources can take some time (e.g. spawning a
+  new VM in the cloud and running ansible playbooks to provision it can take few
+  minutes), and users don't want to wait. It is a good idea to preallocate a
+  small number of resources that are ready to be used immediately.
+- Livechecks - Clouds are unreliable. VMs can break while starting or become
+  unresponsive for various reasons. Resalloc periodically checks the livelines
+  of all resources and makes sure we doesn't leak money.
+- Resource prioritization - Multiple pools can provide the same resources but
+  they might not cost the same. There is a price for running VMs in Amazon
+  AWS. The same VMs as Spot instances are a bit cheaper, and running them on a
+  self-hosted sever is for free. Resalloc prioritizes to be wallet-friendly.
+- Web interface - It is possible to optionally run a web UI and let users see
+  what, and how many resources are available. For example, see
+  [Copr resources](https://download.copr.fedorainfracloud.org/resalloc).
```

Comment:
The list is long enough ... but did I forget something important?

Reply:
This is awesome, thank you very much for the docs!  We can continue in a different PR, but something that comes to my mind:
- resource recycling and security -> I can get the very same resource again for multiple tasks just to not waste the resource allocation time, and potentially fixed price for spawning
- fair client service -> despite the overall complicated logic in allocation, the sooner the client comes the earlier is served
- logging -> each resource has it's own set of "live" logs you may monitor by `tail -f`